### PR TITLE
React to Security/Http changes

### PR DESCRIPTION
**Pull request by HaoK**
https://github.com/aspnet/HttpAbstractions/pull/323
https://github.com/aspnet/Security/pull/283

cc @rynowak

### CODE REVIEW

File: test/WebSites/FiltersWebSite/BasicAuthenticationHandler.cs [hunk 1]
```diff
@@ -32,7 +24,7 @@ protected override AuthenticationTicket AuthenticateCore()
                     new Claim(ClaimTypes.NameIdentifier, "John")
                 },
                 Options.AuthenticationScheme));
-            return new AuthenticationTicket(principal, new AuthenticationProperties(), Options.AuthenticationScheme);
+            return Task.FromResult(new AuthenticationTicket(principal, new AuthenticationProperties(), Options.AuthenticationScheme));
```
Comment (rynowak) [inline test/WebSites/FiltersWebSite/BasicAuthenticationHandler.cs]: Break up long line

File: test/Microsoft.AspNet.Mvc.FunctionalTests/FiltersTest.cs [hunk 2]
```diff
@@ -155,7 +155,7 @@ public class FiltersTest
                 "http://localhost/AuthorizeUser/Impossible");
 
             // Assert
-            Assert.Equal(HttpStatusCode.Unauthorized, response.StatusCode);
+            Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/FiltersTest.cs]: How intended is this behavior change?

Comment (HaoK) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/FiltersTest.cs]: This is actually finally the correct behavior, when a policy fails with a missing claim, you want 403 forbidden (not unauthorized).  This finally fixes it (for one path).  There's still an issue which currently isn't shown in the mvc tests)

Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/FiltersTest.cs]: Ok, cool. I'll sign off on this change, but can you send a PR to add the missing test with `[Fact(Skip = "blah blah")]`. I don't think anyone else is aware of the bug.

**Comment by rynowak** [comment]
:shipit:

File: test/Microsoft.AspNet.Mvc.FunctionalTests/FiltersTest.cs (hunk 2 above)
Comment (HaoK) [inline test/Microsoft.AspNet.Mvc.FunctionalTests/FiltersTest.cs]: Yeah this is tracking the case that fails https://github.com/aspnet/Security/issues/300

**Comment by HaoK** [comment]
7c426f7835731e620f7dd0d03f58bf63ab33d301

**Pull request closed without merging**